src/test/java/org/broadinstitute/hellbender/tools/exome/SubtractCoverageComponentsIntegrationTest.java

```diff
@@ -0,0 +1,117 @@
+
+        final PCA pca;
+        try (final HDF5File pcaFile = new HDF5File(TEST_PCA_FILE, HDF5File.OpenMode.READ_ONLY)) {
+            pca = PCA.readHDF5(pcaFile);
+        }
+        final RealVector centers = pca.getCenters();
+        final List<String> variables = pca.getVariables();
+        final List<Target> targets = variables.stream().map(Target::new).collect(Collectors.toList());
+        final ReadCountCollection inputCounts = ReadCountCollectionUtils.parse(inputFile).arrangeTargets(targets);
+        final ReadCountCollection outputCounts = ReadCountCollectionUtils.parse(outputFile).arrangeTargets(targets);
+        final List<String> samples = inputCounts.columnNames();
+        final RealMatrix eigenVectors = pca.getEigenVectors();
+
+        for (int i = 0; i < samples.size(); i++) {
+            final String sample = samples.get(i);
+            final int outputIndex = outputCounts.columnNames().indexOf(sample);
+            if (outputIndex < 0) {
+                Assert.fail("cannot find sample: " + sample + " in " + Arrays.toString(outputCounts.columnNames().toArray()));
+            }
+            final double[] inputSampleValues = inputCounts.counts().getColumn(i);
+            final double[] centeredSampleValues = new double[inputSampleValues.length];
+            for (int j = 0; j < centeredSampleValues.length; j++) {
+                centeredSampleValues[j] = inputSampleValues[j] - centers.getEntry(j);
+            }
+            final double[] outputSampleValues = outputCounts.counts().getColumn(outputIndex);
+            final double[] expectedSampleValues = centeredSampleValues.clone();
+            final double[] revertedSampleValues = outputSampleValues.clone();
+            for (int k = 0; k < eigenVectors.getColumnDimension(); k++) {
+                final double loading = new ArrayRealVector(eigenVectors.getColumn(k)).dotProduct(new ArrayRealVector(centeredSampleValues));
+                final RealVector projection = new ArrayRealVector(eigenVectors.getColumn(k)).mapMultiply(loading);
+                final RealVector updatedCenteredSampleValues = new ArrayRealVector(centeredSampleValues).subtract(projection);
+                System.arraycopy(updatedCenteredSampleValues.toArray(), 0, centeredSampleValues, 0, centeredSampleValues.length);
+                final RealVector updatedRevertedValues = new ArrayRealVector(revertedSampleValues).add(projection);
+                System.arraycopy(updatedRevertedValues.toArray(), 0, revertedSampleValues, 0, revertedSampleValues.length);
+            }
+            for (int k = 0; k < revertedSampleValues.length; k++) {
+                Assert.assertEquals(revertedSampleValues[k], expectedSampleValues[k], 0.000001, " i and k  = " + i + " " + k);
+            }
+        }
+    }
+
+    @AfterClass
+    protected static void disposeOfTestFile() {
```

Comment:
do you need to explicitly clean things up here?  I think that createTempFile will take care of that for you?